src/auxiliary/special_elixirs.jl

```diff
@@ -145,6 +149,38 @@ convergence_test(elixir::AbstractString, iterations; kwargs...) = convergence_te
 walkexpr(f, expr::Expr) = f(Expr(expr.head, (walkexpr(f, arg) for arg in expr.args)...))
 walkexpr(f, x) = f(x)
 
+# Insert the keyword argument `maxiters` into calls to `solve` and `Trixi.solve`
+# with default value `10^5` if it is not already present.
+function insert_maxiters(expr)
+  maxiters_default = 10^5
+
+  expr = walkexpr(expr) do x
+    if x isa Expr
+      is_plain_solve = x.head === Symbol("call") && x.args[1] === Symbol("solve")
+      is_trixi_solve = (x.head === Symbol("call") && x.args[1] isa Expr &&
+                        x.args[1].head === Symbol(".") &&
+                        x.args[1].args[1] === Symbol("Trixi") &&
+                        x.args[1].args[2] isa QuoteNode &&
+                        x.args[1].args[2].value === Symbol("solve"))
```

Comment:
I have no idea what's going on here. Maybe this should be reviewed by someone who has more experience with expressions?

Reply:
I think that person that understands *anything* about expressions besides @ranocha still has to join the Trixi fold :-)

AFAICT, here for each expression is checked if it is a function call to a plain function (line 159), which would catch calls like
```julia
solve(...)
```
That is, our usual call to OrdinaryDiffEq's `solve`.

The second part is slightly more convoluted, since it needs to capture the calls where we use Trixi's internal solve. Since we do not export it, it always has to be prefixed by `Trixi.` (unless someone is `using` it, which would then fall back to the first case above). Thus, the second pattern matches a "call", where the first argument is again another expression that includes a `.`, and where the first subargument is `Trixi`. I do not fully understand `QuoteNode`, but together I think it will match
```
Trixi.solve(...)
```
Please correct me if I'm wrong, @ranocha !

Reply:
That's correct - you can come up with this stuff by `dump`ing expressions you want to look at.